sycl/doc/extensions/proposed/sycl_ext_oneapi_work_group_static.asciidoc

```diff
@@ -146,73 +143,89 @@ specified explicitly.
 
 [NOTE]
 ====
-If a `work_group_static` object is declared at function scope, the
+If a `work_group_static` object is declared at a block scope, the
 device local memory associated with the object will be identical for all
 usages of that function within the kernel. In cases where a function is called
 multiple times, developers must take care to avoid race conditions (e.g., by
 calling `group_barrier` before and after using the memory).
 ====
 
+Change to SYCL 2020 section `5.9.2 Common address space deduction rules`:
+Namespace scope: if the variable is `work_group_static` object,
+then the variable is assigned to the local address space.
+Otherwise normal rules applies.
+
 SYCL 2020 requires that all global variables accessed by a device function are
 `const` or `constexpr`. This extension lifts that restriction for
 `work_group_static` variables.
 
-[NOTE]
-====
-Since `work_group_static` acts as a view, wrapping an underlying pointer, a
-developer may still choose to declare variables as `const`.
-====
-
 When `T` is a class type or bounded array, the size of the allocation is known
-at compile-time, and a SYCL implementation may embed the size of the allocation
+at compile-time, and a SYCL implementation embeds the size of the allocation
 directly within a kernel. Each instance of `work_group_static<T>` is associated
 with a unique allocation in device local memory.
 
-When `T` is an unbounded array, the size of the allocation is unknown at
-compile-time, and must be communicated to the SYCL implementation via the
-`work_group_static_memory_size` property. Every instance of `work_group_static`
-for which `T` is an unbounded array is associated with a single, shared,
-allocation in device local memory. For example, two instances declared
-as `work_group_static<int[]>` and `work_group_static<float[]>` will be
-associated with the same shared allocation.
-
-If the total amount of device local memory requested (i.e., the sum of
-all memory requested by `local_accessor`, `group_local_memory`,
-`group_local_memory_for_overwrite` and `work_group_static`) exceeds a device's
-local memory capacity (as reported by `local_mem_size`) then the implementation
-must throw a synchronous `exception` with the `errc::memory_allocation` error
-code from the kernel invocation command (e.g. `parallel_for`).
```

Comment:
Why did you remove this paragraph?  Isn't this still relevant even though `T` can no longer be an unbounded array?